sycl/source/detail/event_impl.cpp

```diff
@@ -50,11 +50,31 @@ event_impl::~event_impl() {
 }
 
 void event_impl::waitInternal() const {
-  if (!MHostEvent) {
+  if (!MHostEvent && MEvent) {
     getPlugin().call<PiApiKind::piEventsWait>(1, &MEvent);
+    return;
   }
-  // Waiting of host events is NOP so far as all operations on host device
-  // are blocking.
+
+  while (MState != HES_Complete)
+    ;
+}
+
+void event_impl::setComplete() {
+  if (MHostEvent || !MEvent) {
```

Comment:
When is it possible that MHostEvent is false but MEvent is null ? Shouldn't that be an error ? (I am actually not sure, just trying to understand better)

Reply:
The event handle can be null in case of, for example, alloca command, which is currently synchronous.